test/e2e/test-e2e.js

```diff
@@ -0,0 +1,323 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+var assert = require('assert');
+var util = require('util');
+var _ = require('lodash'); // for _.find. Can't use ES6 yet.
+var cp = require('child_process');
+var semver = require('semver');
+var promisifyAll = require('@google-cloud/common').util.promisifyAll;
+var Debug = require('../..');
+var Debugger = require('../debugger.js');
+
+var CLUSTER_WORKERS = 3;
+
+var FILENAME = 'test/fixtures/fib.js';
+
+var delay = function(delayTimeMS) {
+  return new Promise(function(resolve, reject) {
+    setTimeout(resolve, delayTimeMS);
+  });
+};
+
+describe('@google-cloud/debug end-to-end behavior (allow 60s)', function () {
+  var api;
+
+  var debuggeeId;
+  var projectId;
+  var children = [];
+
+  before(function() {
+    promisifyAll(Debugger);
+    api = new Debugger(new Debug({}));
+  });
+
+  beforeEach(function() {
+    this.timeout(10 * 1000);
+    return new Promise(function(resolve, reject) {
+      var numChildrenReady = 0;
+      var handler = function(a) {
+        if (a[0].length > 0) {
+          reject(new Error('A child reported the following error: ' + a[0]));
+        }
+        if (!debuggeeId) {
+          // Cache the needed info from the first worker.
+          debuggeeId = a[1];
+          projectId = a[2];
+        } else {
+          // Make sure all other workers are consistent.
+          assert.equal(debuggeeId, a[1]);
+          assert.equal(projectId, a[2]);
+          if (debuggeeId !== a[1] || projectId !== a[2]) {
+            reject(new Error('Child debuggee ID and/or project ID' +
```

Comment:
It might be good to add a return after the `reject` here and above since the function continues executing even after the call (it happens to not matter here because even if it does run the resolve the promise will have already rejected).